### PR TITLE
Fixes lp#1552274 - added json output to list-credentials command.

### PR DESCRIPTION
**Pull request by anastasiamac**
Added presentation layer credentials struct. Added tests for json output.

QA:
1. Run 'juju credentials --format json"
[expected] output should have lower-cased representation of credentials with empty properties omitted

### CODE REVIEW

File: cmd/juju/cloud/listcredentials.go [hunk 1]
```diff
@@ -61,8 +61,35 @@ type listCredentialsCommand struct {
 	cloudByNameFunc    func(string) (*jujucloud.Cloud, error)
 }
 
+type CloudCredential struct {
```
Comment (axw) [inline cmd/juju/cloud/listcredentials.go]: comment?

Comment (anastasiamac) [inline cmd/juju/cloud/listcredentials.go]: :+1:

**Comment by anastasiamac** [comment]
$$merge$$

**Comment by anastasiamac** [comment]
intermittent bug # 1623178 TxnPrunerSuite.TestPrunes
$$merge$$

**Pull request merged**